### PR TITLE
1.2.0 pre-release changes

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: buildSrc/src/main/kotlin/splunk.java-conventions.gradle.kts [hunk 1]
```diff
@@ -13,10 +13,10 @@ repositories {
...
 }
 
-val otelVersion = "1.4.0"
-val otelAlphaVersion = "1.4.0-alpha"
-val otelInstrumentationVersion = "1.5.0-SNAPSHOT"
-val otelInstrumentationAlphaVersion = "1.5.0-alpha-SNAPSHOT"
+val otelVersion = "1.4.1"
+val otelAlphaVersion = "1.4.1-alpha"
+val otelInstrumentationVersion = "1.4.0"
+val otelInstrumentationAlphaVersion = "1.4.0-alpha"
```
Comment (mateuszrzeszutek) [inline buildSrc/src/main/kotlin/splunk.java-conventions.gradle.kts:19]: I've reverted to 1.4.0 for now -- we can't release a stable release based on a snapshot. Will revert it back in post-release PR.

**Comment by mateuszrzeszutek** [comment]
CC @theletterf

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,20 @@ and this repository adheres to [Semantic Versioning](https://semver.org/spec/v2.
...
+
+- OpenTelemetry Java SDK and OpenTelemetry Instrumentation for Java dependencies were updated to version 1.4.0.
+
+### Enhancements
+
+- We have added instrumentations for [HikariCP](https://github.com/brettwooldridge/HikariCP)
+  and [Tomcat JDBC](https://tomcat.apache.org/tomcat-8.5-doc/jdbc-pool.html) connection pools. The agent will now
+  collect and export metrics for those two JDBC connection pools.
+- The agent now supports the `OTEL_SERVICE_NAME` environment variable (and `otel.service.name` system property) - you no
+  longer need to use `OTEL_RESOURCE_ATTRIBUTES` to set the service name.
```
Comment (iNikem) [inline CHANGELOG.md:23]: should we mention that this new way is a preferred one?

**Review by iNikem — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,20 @@ and this repository adheres to [Semantic Versioning](https://semver.org/spec/v2.
 
 ## Unreleased
 
+## v1.2.0 - 2021-07-26
+
+### General
+
+- OpenTelemetry Java SDK and OpenTelemetry Instrumentation for Java dependencies were updated to version 1.4.0.
```
Comment (theletterf) [inline CHANGELOG.md:15]:
```suggestion
- OpenTelemetry Java SDK and OpenTelemetry Instrumentation for Java dependencies have been updated to version 1.4.0.
```

File: CHANGELOG.md (hunk 2 above)
Comment (theletterf) [inline CHANGELOG.md:23]:
```suggestion
### Enhancements

- We have added instrumentation for [HikariCP](https://github.com/brettwooldridge/HikariCP)
  and [Tomcat JDBC](https://tomcat.apache.org/tomcat-8.5-doc/jdbc-pool.html) connection pools. The agent now
  collects and exports metrics for both JDBC connection pools.
- You can now set the service name using the `OTEL_SERVICE_NAME` environment variable and the `otel.service.name` system property. This removes the need of using `OTEL_RESOURCE_ATTRIBUTES` to set the service name.
```

**Review by theletterf — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (theletterf) [reply CHANGELOG.md:23]: Discussed in Slack.

**Pull request merged**